.github/workflows/pr_check_workflow.yml

```diff
@@ -1,25 +1,211 @@
+  build-lint-test:
     runs-on: ubuntu-latest
+    name: Build and Verify
+    steps:
+      # Access a cache of set results from a previous run of the job
+      # This is to prevent re-running steps that were already successful since it is not native to github actions
+      # Can be used to verify flaky steps with reduced times
+      - name: Restore the cached run
+        uses: actions/cache@v2
+        with:
+          path: |
+            job_successful
+            linter_results
+            unit_tests_results
+            integration_tests_results
+          key: ${{ github.run_id }}-${{ github.job }}-${{ github.sha }}
+          restore-keys: |
+            ${{ github.run_id }}-${{ github.job }}-${{ github.sha }}
+
+      - name: Get if previous job was successful
+        id: job_successful
+        run: cat job_successful 2>/dev/null || echo 'false'
+
+      - name: Get the previous linter results 
+        id: linter_results
+        run: cat linter_results 2>/dev/null || echo 'default'
+
+      - name: Get the previous unit tests results 
+        id: unit_tests_results
+        run: cat unit_tests_results 2>/dev/null || echo 'default'
+
+      - name: Get the previous integration tests results 
+        id: integration_tests_results
+        run: cat integration_tests_results 2>/dev/null || echo 'default'
+
+      - name: Checkout code
+        if: steps.job_successful.outputs.job_successful != 'true'
+        uses: actions/checkout@v2
+
+      - name: Setup Node
+        if: steps.job_successful.outputs.job_successful != 'true'
+        uses: actions/setup-node@v2
+        with:
+          node-version: "10.24.1"
+          registry-url: 'https://registry.npmjs.org'
+
+      - name: Setup Yarn
+        if: steps.job_successful.outputs.job_successful != 'true'
+        run: |
+          npm uninstall -g yarn
+          npm i -g yarn@1.22.10
+
+      - name: Run bootstrap
+        if: steps.job_successful.outputs.job_successful != 'true'
+        run: yarn osd bootstrap
+
+      - name: Run linter
+        if: steps.linter_results.outputs.linter_results != 'success'
+        id: linter
+        run: yarn lint
+
+      # Runs unit tests while limiting workers because github actions will spawn more than it can handle and crash
+      # Continues on error but will create a comment on the pull request if this step failed.
+      - name: Run unit tests
+        if: steps.unit_tests_results.outputs.unit_tests_results != 'success'
+        id: unit-tests
+        continue-on-error: true
+        run: node scripts/jest --ci --colors --maxWorkers=10
+        env:
+          SKIP_BAD_APPLES: true
+
+      - run: echo Unit tests completed unsuccessfully. However, unit tests are inconsistent on the CI so please verify locally with `yarn test:jest`.
+        if: steps.unit_tests_results.outputs.unit_tests_results != 'success' && steps.unit-tests.outcome != 'success'
+      
+      # TODO: This gets rejected, we need approval to add this
```

Comment:
Yeah I wish I didn't have to add this step at all but at least it gets us one step to having a CI in github.

However there are a few github actions that can add a comment, should we agree to the best one before proposing. Another solution could be we just fail on this step and just have someone continue to re-run until it passes. At least it won't re-run the other steps. I wished github didn't give a green checkmark on `contine-on-error`.

Reply:
Do you want to explain the options so that the team can weigh in? I'm okay with failing on unit tests - that should push us more to eliminate flaky tests.

Reply:
> should push us more to eliminate flaky tests.

I think the flakiness is github runners being purposefully limit something to which I feel like might be out of our control unless we want to migrate to calling to Jenkins.

In terms of github actions, if we want to have something to at least log a message telling people to re-run but not block them we can use (github script)[https://github.com/marketplace/actions/github-script] which is a verified action and it can future proof us if we want to do other things. But it is definitely more than what we need right now. This (one)[https://github.com/marketplace/actions/sticky-pull-request-comment], it comments on the PR and updates the same comment instead of appending a new comment everytime. But it's not verified.

Reply:
Yeah I'm good with getting this out the door and iterating as needed!